bsi_zoo/metrics.py

```diff
@@ -68,7 +57,62 @@ def euclidean_distance(x, x_hat, *args, **kwargs):
             x_hat[active_set_hat], active_set_hat, fwd, tmin=1, tstep=1
         )  # estimate
 
-        # euclidean distance check
+    return stc, stc_hat, fwd
+
+
+def jaccard_error(x, x_hat, *args, **kwargs):
+    # You can read more on the Jaccard score in Scikit-learn definition https://scikit-learn.org/stable/modules/generated/sklearn.metrics.jaccard_score.html
+    return 1 - jaccard_score(x, x_hat, average="samples")
+
+
+def mse(x, x_hat, *args, **kwargs):
+    return mean_squared_error(x, x_hat)
+
+
+def EMD(x, x_hat, *args, **kwargs):
+    orientation_type = kwargs["orientation_type"]
+    subject = kwargs["subject"]
+    nnz = kwargs["nnz"]
+
+    if orientation_type == "fixed":
+        temp = np.linalg.norm(x, axis=1)
+        a = temp[np.nonzero(temp)]
+
+        temp = np.linalg.norm(x_hat, axis=1)
+        temp_ = np.partition(-temp, nnz)
+        b = -temp_[:nnz]  # get n(=nnz) max amplitudes
```

Comment:
you don't need to only select nnz sources here. The EMD can be between vectors of different sizes

Reply:
@agramfort  Yes, I agree.
I'm not sure how to get coordinates arranged in the same position order as x/x_hat.

Reply:
which is why just I selected nnz vertices for now.